apps/podcast-transcribe-episode/src/python/podcast_transcribe_episode/workflow_interface.py

```diff
@@ -0,0 +1,302 @@
+import dataclasses
+from datetime import timedelta
+from typing import Optional
+
+# noinspection PyPackageRequirements
+from temporal.activity_method import activity_method, RetryParameters
+# noinspection PyPackageRequirements
+from temporal.workflow import workflow_method
+
+from mediawords.workflow.exceptions import McPermanentError
+
+from .config import PodcastTranscribeEpisodeConfig
+from .enclosure import StoryEnclosureDict
+from .media_info import MediaFileInfoAudioStreamDict
+
+TASK_QUEUE = "podcast-transcribe-episode"
+"""Temporal task queue."""
+
+DEFAULT_RETRY_PARAMETERS = RetryParameters(
+
+    # InitialInterval is a delay before the first retry.
+    initial_interval=timedelta(seconds=1),
+
+    # BackoffCoefficient. Retry policies are exponential. The coefficient specifies how fast the retry interval is
+    # growing. The coefficient of 1 means that the retry interval is always equal to the InitialInterval.
+    backoff_coefficient=2,
+
+    # MaximumInterval specifies the maximum interval between retries. Useful for coefficients more than 1.
+    maximum_interval=timedelta(hours=2),
+
+    # MaximumAttempts specifies how many times to attempt to execute an Activity in the presence of failures. If this
+    # limit is exceeded, the error is returned back to the Workflow that invoked the Activity.
+
+    # We start off with a huge default retry count for each individual activity (1000 attempts * 2 hour max. interval
+    # = about a month worth of retrying) to give us time to detect problems, fix them, deploy fixes and let the workflow
+    # system just handle the rest without us having to restart workflows manually.
+    #
+    # Activities for which retrying too much doesn't make sense (e.g. due to the cost) set their own "maximum_attempts".
+    maximum_attempts=1000,
+
+    # NonRetryableErrorReasons allows you to specify errors that shouldn't be retried. For example retrying invalid
+    # arguments error doesn't make sense in some scenarios.
+    non_retryable_error_types=[
+
+        # Counterintuitively, we *do* want to retry not only on transient errors but also on programming and
+        # configuration ones too because on programming / configuration bugs we can just fix up some code or
+        # configuration, deploy the fixes and let the workflow system automagically continue on with the workflow
+        # without us having to dig out what exactly has failed and restart things.
```

Comment:
🤯

Reply:
That's what Temporal's CEO suggested in one of the forum threads - just retry things until you get around to fixing the code that tries to do the thing :) We'll see if this approach is effective for us.